### ISSUE TITLE
await callSync in test hangs for commit_entry of private entry

### ISSUE THREAD

**Issue by jmday** [holochain/holochain-rust#1318]
The following test code hangs when attempting to create a private entry.

in `test/index.js`
```
  const addr = await alice.callSync("message_zome", "create_message", { message_string })
```

in `zomes/message_zome/code/src/lib.rs`
```
pub fn handle_create_message(message_string: String) -> ZomeApiResult<Address> {
    let entry = Entry::App("message_entry".into(), MessageEntry { content: message_string }.into());
    let address = hdk::commit_entry(&entry)?;

    Ok(address)
}
```

Changing the line from `test/index.js` to the following alleviates the issue.
```
  const addr = alice.call("message_zome", "create_message", { message_string })
```

I realize that committing a private entry only requires a write to the local source chain and does not attempt to broadcast the entry to the DHT for storage/replication. However, the use of `await ...callSync` will be prevalent among developers and these hanging tests without any error output is probably not an appropriate result for when this situation arises. :)

For a sample hApp that provides a useful test case, please see:
[https://github.com/holochain/core-concepts-examples/tree/02-HelloMe](https://github.com/holochain/core-concepts-examples/tree/02-HelloMe)
and change the following line to use `await alice.callSync`
[https://github.com/holochain/core-concepts-examples/blob/e0d0cdfa741c66af3be145b95d9f7895c7b572c5/test/index.js#L25](https://github.com/holochain/core-concepts-examples/blob/e0d0cdfa741c66af3be145b95d9f7895c7b572c5/test/index.js#L25)

**Comment by Connoropolous** [comment]
thanks for logging like this @jmday . great stuff. We'll get this addressed

**Comment by maackle** [comment]
This was due to faulty Waiter logic but should be fixed by the new Consistency Signals logic (#1431)
